include/drivers/dma.h

```diff
@@ -98,25 +99,28 @@ struct dma_block_config {
 /**
  * @brief DMA configuration structure.
  *
- *     dma_slot             [ 0 : 5 ]   - which peripheral and direction
+ *     dma_slot             [ 0 : 6 ]   - which peripheral and direction
  *                                        (HW specific)
- *     channel_direction    [ 6 : 8 ]   - 000-memory to memory, 001-memory to
- *                                        peripheral, 010-peripheral to memory,
+ *     channel_direction    [ 7 : 9 ]   - 000-memory to memory,
+ *                                        001-memory to peripheral,
+ *                                        010-peripheral to memory,
+ *                                        011-peripheral to peripheral,
  *                                        ...
- *     complete_callback_en [ 9 ]       - 0-callback invoked at completion only
+ *     complete_callback_en [ 10 ]       - 0-callback invoked at completion only
  *                                        1-callback invoked at completion of
  *                                          each block
- *     error_callback_en    [ 10 ]      - 0-error callback enabled
+ *     error_callback_en    [ 11 ]      - 0-error callback enabled
  *                                        1-error callback disabled
- *     source_handshake     [ 11 ]      - 0-HW, 1-SW
- *     dest_handshake       [ 12 ]      - 0-HW, 1-SW
- *     channel_priority     [ 13 : 16 ] - DMA channel priority
- *     source_chaining_en   [ 17 ]      - enable/disable source block chaining
+ *     source_handshake     [ 12 ]      - 0-HW, 1-SW
+ *     dest_handshake       [ 13 ]      - 0-HW, 1-SW
+ *     channel_priority     [ 14 : 17 ] - DMA channel priority
+ *     source_chaining_en   [ 18 ]      - enable/disable source block chaining
  *                                        0-disable, 1-enable
- *     dest_chaining_en     [ 18 ]      - enable/disable destination block
+ *     dest_chaining_en     [ 19 ]      - enable/disable destination block
  *                                        chaining.
  *                                        0-disable, 1-enable
- *     reserved             [ 19 : 31 ]
+ *     linked_channel       [ 20 : 26 ] - linked channel number
```

Comment:
What is this exactly, how is it used and does it apply to all DMA drivers?

Reply:
in imx.rt we can interlink channel, and as there are 127 channels so we need 7 bits for this link_channel settings.  Ohter drivers will not use it as this is used to be a reserved bits.